stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -965,16 +981,42 @@ public func _tffunc<State : _TensorArrayProtocolEnhanced,
   with state: State,
   in fn: @escaping (State, Data) -> State
 ) -> (Data) -> (String) {
-  let wrappedFn = {
-    // The result argument needs to be a type that conforms to TensorGroup.
-    // We are arbitrarily picking Tensor<Float> here.
-    (s: State, d: Data) -> (State, Tensor<Float>?) in
-      (fn(s, d), nil)
-  }
-  let traceContext = _trace(with: state, in: wrappedFn)
+  let wrappedFn = {(inputs: [CTensorHandle]) -> [CTensorHandle] in
+    let symbolicState = state._makeInstance(
+      owning: inputs.dropLast(Data._typeList.count))
+    let symbolicData = Data(
+      _copying: inputs.dropFirst(Int(state._tensorHandleCount)))
+    let outputState = fn(symbolicState, symbolicData)
+    return outputState.cTensorHandles
+  }
+  let dtypes = state._dtypes + Data._typeList.map { $0.cDataType }
+  let traceContext = _trace(with: dtypes, in: wrappedFn)
   return {
-    data in traceContext.specializeTFFunction(with: data)
+    data in traceContext.specializeTFFunction(with: data.cTensorHandles)
+  }
+}
+
+/// Trace the given function and return a `TF_Function(In)` that returns `Out`.
+public func _tffunc<In : TensorGroup, Out : TensorGroup>(
```

Comment:
This returns a String which is the TF function name? It's not clear from the comment.

Reply:
Done.